simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1002 @@
+                hist_values.append(self.hist_time_altitude[step].view().T)
+            x_edges.append(mini_hist.axes.edges[0].flatten())
+            y_edges.append(mini_hist.axes.edges[1].flatten())
+
+        return np.array(x_edges), np.array(y_edges), np.array(hist_values)
+
+    def get_2D_photon_position_distr(self, density=True):
+        """
+        Get 2D histograms of position of the Cherenkov photons on the ground. If density is True,
+        it returns the photon density per square meter.
+
+        Parameters
+        ----------
+        density: bool
+            If True, returns the density distribution. If False, returns the distribution of counts.
+
+        Returns
+        -------
+        numpy.array
+            The edges of the direction histograms in x.
+        numpy.array
+            The edges of the direction histograms in y
+        numpy.ndarray
+            The values (counts) of the histogram.
+
+        Raises
+        ------
+        TypeError:
+            if density is not of type bool.
+        HistogramNotCreated:
+            if the histogram was not previously created.
+        """
+        if density is True:
+            return self._get_2D("density")
+        else:
+            return self._get_2D("counts")
+
+    def get_2D_photon_direction_distr(self):
+        """
+        Get 2D histograms of incoming direction of the Cherenkov photons on the ground.
+
+        Returns
+        -------
+        numpy.array
+            The edges of the direction histograms in cos(x).
+        numpy.array
+            The edges of the direction histograms in cos(y)
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+        return self._get_2D("direction")
+
+    def get_2D_photon_time_altitude(self):
+        """
+        Get 2D histograms of the time and altitude of the photon production.
+
+        Returns
+        -------
+        numpy.array
+            The edges of the direction histograms in ns.
+        numpy.array
+            The edges of the direction histograms in km.
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+        return self._get_2D("time_altitude")
+
+    def get_2D_num_photons_distr(self):
+        """
+        Get the distribution of Cherenkov photons per event per telescope.
+
+        Parameters
+        ----------
+        bins: float
+            Number of bins for the histogram.
+        range: 2-tuple
+            Tuple to define the range of the histogram.
+
+        Returns
+        -------
+        numpy.array
+            Number of photons per event per telescope.
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+        num_of_photons_per_event_per_telescope = np.array(
+            self.get_num_photons_per_event_per_telescope()
+        )
+        num_events_array = np.arange(self.num_events + 1)
+        telescope_indices_array = np.arange(self.num_telescopes + 1)
```

Comment:
in my opinion yes, because it is simply the number of photons, i.e., not the entire list of events. This should not take any computational time.

But if you think it makes more sense to stick to the telescopes given by the indices I could change it.

Reply:
The question is if you do not get the wrong answer in case the user provided indices for specific telescopes. I agree that it has no computational impact.

Reply:
I see your point. The value(s) passed in `self.telescope_indices` are actually intended to be related to the histograms.
The `get_2D_num_photons_distr` function can be used even without creating and filling histograms. This is supposed to represent the corsika output file (i.e. the class instance). 

I think that if we focus only on the telescopes given by `self.telescope_indices` it would lead to confusion. The variables needed for using `get_2D_num_photons_distr` are defined in the `__init__`, which is ran only once. The current implementation allows the user to redefine `self.telescope_indices` and produce new histograms, but in that case  `get_2D_num_photons_distr` would remain the same. Therefore, I prefer to keep it like it is.